src/codegen/mod.rs

```diff
@@ -1,12 +1,123 @@
-use crate::target::Target;
+pub mod c_utils;
+mod clang;
+mod cpu;
+mod header;
+mod namegen;
+
+use crate::codegen::clang::clang_path;
+use crate::codegen::cpu::CpuCodeGenerator;
+use crate::color::do_color;
+use crate::imp::Impl;
+use crate::imp::ImplNode;
+use crate::target::{Target, Targets, X86MemoryLevel};
+use crate::utils::ToWriteFmt;
+use crate::views::Tensor;
 
+use anyhow::{bail, Result};
 use std::fmt;
+use std::fmt::Debug;
+use std::io;
+use std::path::PathBuf;
+use std::process::{Command, Output};
+use std::rc::Rc;
+use tempfile::tempdir;
 
-mod c_utils;
-mod header;
-mod namegen;
-mod x86;
+const CLI_FLAGS: [&str; 3] = ["-std=gnu99", "-O3", "-o"];
+
+const X86_CLI_VEC_FLAGS: [&str; 2] = ["-fopenmp", "-mavx2"];
+const ARM_CLI_VEC_FLAGS: [&str; 1] = ["-fopenmp"];
 
 pub trait CodeGen<Tgt: Target> {
-    fn emit_kernel<W: fmt::Write>(&self, out: &mut W) -> fmt::Result;
+    fn compiler_path() -> Result<String> {
+        clang_path() // Clang by default
```

Comment:
Remove comment. The purpose of a default implementation calling `clang_path()` is already clear.